homeassistant/components/overkiz/climate_entities/hitachi_air_to_air_heat_pump.py

```diff
@@ -0,0 +1,384 @@
+        ) and state.value_as_int:
+            return int(state.value_as_int)
+
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new temperature."""
+        temperature = cast(float, kwargs.get(ATTR_TEMPERATURE))
+        await self._global_control(target_temperature=int(temperature))
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if (
+            self.protocol == Protocol.OVP
+            and (state := self.device.states["core:HolidaysModeState"])
+            and state.value_as_str
+        ):
+            if state.value_as_str == OverkizCommandParam.ON:
+                return PRESET_HOLIDAY_MODE
+
+            if state.value_as_str == OverkizCommandParam.OFF:
+                return PRESET_NONE
+
+        if (
+            self.protocol == Protocol.HLRR_WIFI
+            and (state := self.device.states[LEAVE_HOME_STATE[self.protocol]])
+            and state.value_as_str
+        ):
+            if state.value_as_str == OverkizCommandParam.ON:
+                return PRESET_HOLIDAY_MODE
+
+            if state.value_as_str == OverkizCommandParam.OFF:
+                return PRESET_NONE
+
+        return None
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        # OVP uses setHolidays, where HLRRWIFI uses an extra parameter in global control
+        if self.protocol == Protocol.HLRR_WIFI:
+            if preset_mode == PRESET_HOLIDAY_MODE:
+                await self._global_control(leave_home=OverkizCommandParam.ON)
+
+            if preset_mode == PRESET_NONE:
+                await self._global_control(leave_home=OverkizCommandParam.OFF)
+
+        if self.protocol == Protocol.OVP:
+            if preset_mode == PRESET_HOLIDAY_MODE:
+                await self.executor.async_execute_command(
+                    "setHolidays", OverkizCommandParam.ON  # add to OverkizCommand
+                )
+            if preset_mode == PRESET_NONE:
+                await self.executor.async_execute_command(
+                    "setHolidays", OverkizCommandParam.OFF  # add to OverkizCommand
+                )
+
+    async def _global_control(
+        self,
+        main_operation: str | None = None,
+        target_temperature: int | None = None,
+        fan_mode: str | None = None,
+        hvac_mode: str | None = None,
+        swing_mode: str | None = None,
+        leave_home: str | None = None,
+    ) -> None:
+        """Execute globalControl command with all parameters. There is no option to only set a single parameter, without passing al other values."""
+        main_operation = (
+            main_operation
+            or str(
+                cast(
+                    State, self.device.states[MAIN_OPERATION_STATE[self.protocol]]
+                ).value_as_str
+            ).lower()
```

Comment:
I am not a huge fan of this cast, value_as_str followed by str(). This seems to do the same thing multiple times, can this not be simplified?

Reply:
Not sure about separating OVP and HLLR_WIFI protocols in different entities because they seem quite similar.

I agree about the excess casts, which aren't the proper way of dealing with Optionals. I have a small refactor in commit #07cb23d to make explicit backfill and fallback of missing parameters.